sklearn/gaussian_process/kernels.py

```diff
@@ -1348,9 +1357,9 @@ def __repr__(self):
                 self.__class__.__name__,
                 ", ".join(map("{0:.3g}".format, self.length_scale)),
                 self.nu)
-        else:  # isotropic
+        else:
             return "{0}(length_scale={1:.3g}, nu={2:.3g})".format(
-                self.__class__.__name__, self.length_scale, self.nu)
+                self.__class__.__name__, _squeeze(self.length_scale), self.nu)
```

Comment:
`np.ravel(length_scale)[0]` or `np.atleast_1d(length_scale)[0]` should also work. I think the helper function is a bit obscure to the reader.